db/errors.go

```diff
@@ -29,3 +30,15 @@ func IsDocumentLimit(err error) bool {
 	}
 	return strings.Contains(errors.Cause(err).Error(), "an inserted document is too large")
 }
+
+func IsErrorCode(err error, errorCode int) bool {
+	var mongoError mongo.CommandError
+	if errors.As(err, &mongoError) && mongoError.HasErrorCode(errorCode) {
+		return true
+	}
+	return false
+}
+
+// ErrorCode constants for mongo errors
+
+const FACET_PIPELINE_STAGE_TOO_LARGE_CODE = 4031700 // https://github.com/mongodb/mongo/blob/a1732172ed5d66d98582ea1059c0ede9d8cd5065/src/mongo/db/pipeline/document_source_facet.cpp#L165
```

Comment:
nit: Go style in Evergreen uses `PascalCase` for constants rather than C-style `YELLING_CASE`.